### PR TITLE
Add sample APIs for express mongoose app

### PR DESCRIPTION
**Pull request by Frey0-0**
Fixes [#490](https://github.com/keploy/keploy/issues/490)
- I have added sample APIs for mongoose 
- Used mongoose version `6.10.3` because there are some breaking changes in the latest version
- Used the mongoose API

### CODE REVIEW

File: express-mongoose/server.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    newUser.save()
+    .then(data => JSON.stringify(data))
+    .then(text => rs.send(text));
+});
+
+// Patch route to update data in MongoDB
+app.patch('/updateData', function (re, rs) {
+    console.log("PATCH request at /updateData route");
+    console.log(re.body);
+    user.findOneAndUpdate({name: re.body.name}, re.body)
```
Comment (re-Tick) [inline express-mongoose/server.js:38]: Please use the supported mongoose updateOne method to  mock the outputs of the dependency call.

File: express-mongoose/server.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    .then(data => JSON.stringify(data))
+    .then(text => rs.send(text));
+});
+
+// Delete route to delete data from MongoDB
+app.delete('/deleteData', function (re, rs) {
+    console.log("DELETE request at /deleteData route");
+    console.log(re.body);
+    user.findOneAndDelete({name: re.body.name})
```
Comment (re-Tick) [inline express-mongoose/server.js:47]: Please use deleteOne mongoose method to mock the outputs.

File: express-mongoose/server.js (hunk 1 above)
Comment (Frey0-0) [reply express-mongoose/server.js:38]: Yes I have changed, the updateOne is implemented by the `updateOne` method of the mongodb driver whereas the findOneAndUpdate uses the `findAndModify`

File: express-mongoose/server.js (hunk 2 above)
Comment (Frey0-0) [reply express-mongoose/server.js:47]: Yes I will change this as well.

**Comment by Frey0-0** [comment]
I have changed the methods and re-recorded the tests as well and now the proper method are stored in the mock yaml files as well :smile:

**Comment by Sarthak160** [comment]
Hey @Frey0-0 , Can you add steps to generate tests and mocks as we have done in node-fetch application. It will be super helpful .

**Comment by Frey0-0** [comment]
Yup for sure, I will do it some time.

**Comment by Frey0-0** [comment]
Hey @Sarthak160 I have made the changes, sorry it took some time I was ill for the past couple of days :sweat:

**Comment by Sarthak160** [comment]
> Hey @Sarthak160 I have made the changes, sorry it took some time I was ill for the past couple of days 😓

No problem 😇 , Hope you are good now . Btw you can also add a docker command or a docker-compose file to run mongo instance as well .

**Comment by Frey0-0** [comment]
Oh yes, that will remove the possibility of absence of db.  I will add a docker-compose file and the docker command in the readme

**Comment by Frey0-0** [comment]
Hey @Sarthak160, please have a look and tell if there are any more changes required :smiley:

**Comment by Frey0-0** [comment]
@Sarthak160 ping

**Review by Sarthak160 — APPROVED** [review]
LGTM

**Pull request merged**